chipsec/helper/linuxnative/linuxnativehelper.py

```diff
@@ -0,0 +1,462 @@
+        Returns None if no region matches.
+        """
+        for region in self.mappings:
+            if region.start <= base and region.end >= base + size:
+                return region
+        return None
+
+    def map_io_space(self, base: int, size: int, cache_type: int) -> None:
+        """Map to memory a specific region."""
+        if self.devmem_available() and not self.memory_mapping(base, size):
+            logger().log_debug(f"[helper] Mapping 0x{base:x} to memory")
+            length = max(size, resource.getpagesize())
+            page_aligned_base = base - (base % resource.getpagesize())
+            mapping = MemoryMapping(self.dev_mem, length, mmap.MAP_SHARED,
+                                    mmap.PROT_READ | mmap.PROT_WRITE,
+                                    offset=page_aligned_base)
+            self.mappings.append(mapping)
+
+
+    def read_phys_mem(self, phys_address_hi: int, phys_address_lo: int, length: int) -> bytes:
+        phys_address = (phys_address_hi << 32) | phys_address_lo
+        if self.devmem_available():
+            os.lseek(self.dev_mem, phys_address, os.SEEK_SET)
+            return os.read(self.dev_mem, length)
+        return b'\x00'
+
+    def write_phys_mem(self, phys_address_hi: int, phys_address_lo: int, length: int, newval: bytes) -> int:
+        phys_address = (phys_address_hi << 32) | phys_address_lo
+        if newval is None:
+            return None
+        if self.devmem_available():
+            os.lseek(self.dev_mem, phys_address, os.SEEK_SET)
+            written = os.write(self.dev_mem, newval)
+            if written != length:
+                logger().log_debug(f"Cannot write {newval} to memory {phys_address:016X} (wrote {written:d} of {length:d})")
+            return written
+        return -1
+
+    def alloc_phys_mem(self, length, max_phys_address):
+        raise NotImplementedError()
+
+    def free_phys_mem(self, physical_address):
+        raise NotImplementedError()
+
+    def va2pa(self, va):
+        raise NotImplementedError()
+
+    def read_io_port(self, io_port: int, size: int) -> int:
+        if self.devport_available():
+            os.lseek(self.dev_port, io_port, os.SEEK_SET)
+
+            value = os.read(self.dev_port, size)
+            if 1 == size:
+                return struct.unpack("B", value)[0]
+            elif 2 == size:
+                return struct.unpack("H", value)[0]
+            elif 4 == size:
+                return struct.unpack("I", value)[0]
+        return -1
+
+    def write_io_port(self, io_port: int, value: int, size: int) -> bool:
+        if self.devport_available():
+            os.lseek(self.dev_port, io_port, os.SEEK_SET)
+            if 1 == size:
+                fmt = 'B'
+            elif 2 == size:
+                fmt = 'H'
+            elif 4 == size:
+                fmt = 'I'
+            else:
+                raise ValueError("Invalid size")
```

Comment:
read condition does not have an else case that raises ValueError, suggest making it consistent

Reply:
Read condition for io_port (`read_io_port:318`) does have similar condition. Were you referring to the `read_pci_reg` function?